code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,823 @@
+	RegisterSignal(src, COMSIG_PARENT_PREQDELETED, PROC_REF(deleted_handler))
+
+	RegisterSignal(SSdcs, COMSIG_GLOB_CABLE_UPDATED, PROC_REF(cable_updated_handler))
+
+	current_power = locate(/obj/machinery/power) in loc
+	// in the case that both current_power and current_cable are null, the pulsedemon will die the next tick
+	if(!current_power)
+		current_cable = locate(/obj/structure/cable) in loc
+	else
+		forceMove(current_power)
+	update_glow()
+	playsound(get_turf(src), 'sound/effects/eleczap.ogg', 30, TRUE)
+	give_spells()
+
+/mob/living/simple_animal/demon/pulse_demon/proc/deleted_handler(src, force)
+	SIGNAL_HANDLER
+	// assume normal deletion if we're on a turf, otherwise deletion could be inherited from loc
+	if(force || isnull(loc) || isturf(loc))
+		return FALSE
+	// if we did actually die, simple_animal/death will set del_on_death to FALSE before calling qdel
+	if(!del_on_death)
+		return FALSE
+	exit_to_turf()
+	return TRUE
+
+/mob/living/simple_animal/demon/pulse_demon/proc/cable_updated_handler(SSdcs, turf/T)
+	SIGNAL_HANDLER
+	if(cable_images[T])
+		var/list/turf_images = cable_images[T]
+		for(var/image/current_image in turf_images)
+			client?.images -= current_image
+		turf_images.Cut()
+	else
+		cable_images[T] = list()
+
+	for(var/obj/structure/cable/C in T)
+		var/image/cable_image = image(C, C, layer = ABOVE_LIGHTING_LAYER, dir = C.dir)
+		cable_image.plane = ABOVE_LIGHTING_PLANE
+		cable_images[T] += cable_image
+		client?.images += cable_image
+
+/mob/living/simple_animal/demon/pulse_demon/proc/apc_deleted_handler(obj/machinery/power/apc/A, force)
+	SIGNAL_HANDLER
+	hijacked_apcs -= A
+
+/mob/living/simple_animal/demon/pulse_demon/Destroy()
+	cable_images.Cut()
+	apc_images.Cut()
+
+	controlling_area = null
+	current_bot = null
+	current_cable = null
+	current_power = null
+	current_robot = null
+	current_weapon = null
+	apc_being_hijacked = null
+	hijacked_apcs = null
+	hijacked_robots = null
+
+	return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/Login()
+	. = ..()
+	update_cableview()
+
+/mob/living/simple_animal/demon/pulse_demon/vv_edit_var(var_name, var_value)
+	switch(var_name)
+		if("glow_color")
+			update_glow()
+		if("charge")
+			// automatically adjusts maxcharge to allow the new value
+			adjust_charge(var_value - charge, TRUE)
+			return TRUE
+	return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/forceMove(atom/destination)
+	var/old = loc
+	. = ..()
+	current_weapon = null
+	current_robot = null
+	if(current_bot)
+		current_bot.hijacked = FALSE
+	current_bot = null
+	if(old && istype(old, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = old
+		// only set rigged if there are no remaining demons in the cell
+		C.rigged = !(locate(/mob/living/simple_animal/demon/pulse_demon) in old)
+	if(loc && istype(loc, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = loc
+		C.rigged = FALSE
+
+/mob/living/simple_animal/demon/pulse_demon/proc/give_objectives()
```

Comment:
Don't we have better ways of dealing with this? I can understand not wanting to dig into antag datums, but this feels odd and pretty snowflakey

Reply:
probably but I don't wanna touch this to be entirely honest